operator/controllers/loki/rulerconfig_controller.go

```diff
@@ -32,6 +32,12 @@ func (r *RulerConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	key := client.ObjectKey{Name: req.Name, Namespace: req.Namespace}
 	if err := r.Get(ctx, key, &rc); err != nil {
 		if errors.IsNotFound(err) {
+			// RulerConfig not found, remove annotation from LokiStack.
+			err = lokistack.RemoveRulerConfigAnnotation(ctx, r.Client, req.Name, req.Namespace)
```

Comment:
This would be an issue if we have `finalizers` defined for this object, but we don't. So in this case, the object is marked for deletion and is immediately deleted.

Reply:
I was under the impression that checking for `deletionTimestamp` also worked without finalizers and thus was the preferred way. But actually testing the current code, it seems to work properly. Sorry, seems I remembered wrong. :slightly_smiling_face: